### DISCUSSION TITLE
Device Status Query

### DISCUSSION THREAD

**Discussion by gegatesSC** [dgreif/ring#1021] [Q&A]
Hello,

I'm using the ring-client-api in combo with homebridge.  Wondering if you could help - I'm trying to find the status of a device; specifically a ring lighting transformer.  The API creates a light status (simple binary) and a charging status (don't believe it has a battery).  I'm trying to monitor this device if it loses power.  When the power is removed I still have {StatusLowBattery: 0, BatteryLevel: 100, ChargingState: 1} from the accessory.  

Is there a way to get the device health that is shown in the Ring App?  I can see the device goes to offline in the app but there doesn't appear to be a way to connect this to homebridge.

I have a critical pump that runs on the same circuit, and this helps me monitor breaker trips.  Thanks!

**Comment by dgreif** [comment]
All of the data should be on `device.data`. As far as I know, that's the exact same data being used by the Ring app. I'd suggest looking at it again and see if there is a status fields or something of the sort that may indicate the device is offline

**Comment by magedadel** [comment]
I know this is a bit old, but I'm still looking for the contact sensor status, any luck?
I've looked at all the data coming back from the getDevices method and nothing is clear of the status, maybe we need to call another method on the device?

**Reply by tsightler** [reply]
getDevices isn't really designed to get state, rather it gets a list of devices, although it does include current state data in the `device.data` property.  For real time state updates you should subscribe to `device.onData`.  Then each time state data for the device is updated, you can process any property changes.  There are plenty of code examples on contact sensors, you want the `device.data.faulted` property if you are simply trying to determine if the contact sensor is open or closed.

**Reply by magedadel** [reply]
Thank you so much, that did it for me.

**Comment by geetee24** [comment]
do we have to pole every 2 seconds to get real time status?  Or is there a pub sub webhook?

**Reply by magedadel** [reply]
Hi Geetee, like @tsightler mentioned, you should look at `device.data.faulted` to get the current state and then subscribe to `device.onData` for any changes.
